dedupe/datamodel.py

```diff
@@ -183,30 +178,33 @@ def typify_variables(
         try:
             variable_class = VARIABLE_CLASSES[variable_type]
         except KeyError:
+            valid = ", ".join(VARIABLE_CLASSES)
             raise KeyError(
-                "Field type %s not valid. Valid types include %s"
-                % (definition["type"], ", ".join(VARIABLE_CLASSES))
+                f"Variable type {variable_type} not valid. Valid types include {valid}"
             )
-
         variable_object = variable_class(definition)
-        assert isinstance(variable_object, FieldVariable)
-
-        primary_variables.append(variable_object)
-
-        if hasattr(variable_object, "higher_vars"):
-            all_variables.extend(variable_object.higher_vars)
-        else:
-            variable_object = cast(Variable, variable_object)
-            all_variables.append(variable_object)
+        assert isinstance(variable_object, Variable)
+        variables.append(variable_object)
 
+    only_custom = all(isinstance(v, (CustomType, InteractionType)) for v in variables)
```

Comment:
variable name should change. maybe `no_blocking_variables`

Reply:
Responded in followup pr #1102